### ISSUE TITLE
dashboard command gives warning instead of opening dashboard page

### ISSUE THREAD

**Issue by Anomalocaridid** [atlas-engineer/nyxt#2056]
**Describe the bug**
When I run the `dashboard` command, I get the message `Warning: Error on separate thread: The value of STRING is NIL, which is not of type STRING` instead of opening the dashboard page.

**Precise recipe to reproduce the issue**
1. Open Nyxt
2. Run the command `dashboard`
3. Nothing will happen, but the aforementioned message will appear in the message buffer.

For website-specific issues:
Can you reproduce this issue with Epiphany / GNOME Web (https://wiki.gnome.org/Apps/Web)?

This is not a website-specific issue.

**Information**
- OS name+version: EndeavourOS Rolling
- Graphics card and driver: AMD Radeon RX 580 and AMDGPU
- Desktop environment / Window manager name+version: Sway 1.7
- How you installed Nyxt (Guix pack, package manager, build from source): nyxt AUR package installed using paru
- Information from `show-system-information`: 

Output from `nyxt --system-information`:

```
Nyxt version: 2.2.4
Renderer version: GI-GTK
Operating system kernel: Linux 5.16.2-arch1-1
Lisp implementation: SBCL 2.1.11 (Dynamic space size: 3221225472)
Features: (:WEBKIT2 :WEBKIT2-2.34 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL :21BIT-CHARS
 :CHUNGA :CLOSER-MOP :CL-PPCRE-UNICODE :FLEXI-STREAMS :CL-UNICODE :CL-PPCRE
 :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED
 :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS
 :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CHIPZ-SYSTEM:GRAY-STREAMS
 CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI
 CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO
 :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON :THREAD-SUPPORT :ASDF3.3
 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P
 :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF
 :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES
 :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL
 :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY)
Critical dependencies: (/home/anomalocaris/.cache/paru/clone/nyxt/src/nyxt/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /home/anomalocaris/.cache/paru/clone/nyxt/src/nyxt/_build/cl-gobject-introspection/cl-gobject-introspection.asd
 /home/anomalocaris/.cache/paru/clone/nyxt/src/nyxt/_build/cl-webkit/webkit2/cl-webkit2.asd)
```

  If you can't run `show-system-information`, try `nyxt --system-information` from
  a shell.  It this still does not work, please provide the following:
  - Nyxt version (from the `nyxt-version` command or `nyxt --version` from a shell):
  - Lisp implementation/version (if built from source):
  - Kernel name+version:
  - WebKitGTK+ | QtWebEngine version:

**Output when started from a shell**
```
<INFO> [21:16:39] Listening to socket "/run/user/1000/nyxt/nyxt.socket".
<INFO> [21:16:39] Loading Lisp file "/home/anomalocaris/.config/nyxt/auto-config.lisp".
<INFO> [21:16:39] Loading Lisp file "/home/anomalocaris/.config/nyxt/init.lisp".
<INFO> [21:16:39] Not restoring session.
<INFO> [21:16:39] Loading history of 32 URLs from "/home/anomalocaris/.local/share/nyxt/history/default.lisp".
<INFO> [21:16:40] Deleting socket "/run/user/1000/nyxt/nyxt.socket".
```

**Comment by aartaka** [comment]
Do you have any bookmarks? Does adding one solve the problem?

If so, then it's a known bug, and it was already fixed on master :)

**Comment by Anomalocaridid** [comment]
Adding a bookmark solves the problem.

Although it looks like the page itself is kind of messed up. A line goes straight through the four buttons between the bookmarks and the date and time, which occurs even if I run `nyxt -I` and run the dashboard command.

It also seems to override some of the style that I set for the `internal-buffer` class, specifically the "browser☺" part of the title, the horizontal lines between sections, and the section headings.

**Comment by aartaka** [comment]
236e8cc89766a86b15ef4509abc0b1503d44855d should simplify the layout. Closing this now that both bookmark issue is resolved (#2051) and layout is most probably non-breaking (236e8cc89766a86b15ef4509abc0b1503d44855d)?

**Comment by Anomalocaridid** [comment]
Thank you. It looks fine now. Sorry for taking so long to reply. I used the non-git AUR package for Nyxt initially and there seems to have been a few breaking changes between the latest release and the latest commit. It took a while for me to figure out how to fix my config so I could make sure the dashboard worked fine with it.
